src/components/content/observations/patient-observations.tsx

```diff
@@ -1,16 +1,24 @@
 import cx from "classnames";
+import { useEffect } from "react";
 import { PatientObservationsBase } from "@/components/content/observations/helpers/base";
 import { withErrorBoundary } from "@/components/core/error-boundary";
 import { usePatientDiagnosticReports } from "@/fhir/diagnostic-report";
+import { Telemetry } from "@/utils/telemetry";
 
 export type PatientObservationsProps = {
   className?: cx.Argument;
 };
 
 const Component = ({ className }: PatientObservationsProps) => {
-  const diagnosticReports = usePatientDiagnosticReports();
+  const query = usePatientDiagnosticReports();
 
-  return <PatientObservationsBase className={className} query={diagnosticReports} />;
+  useEffect(() => {
+    if (!query.isLoading) {
+      Telemetry.reportZAPRecordCount("builder_observations", query.data?.length);
+    }
+  }, [query.isLoading, query.data]);
```

Comment:
same here (checking query.data exists before calling reportZAPRecordCount

```suggestion
  useEffect(() => {
    if (!query.isLoading && query.data) {
      Telemetry.reportZAPRecordCount("builder_observations", query.data.length);
    }
  }, [query.isLoading, query.data]);
```